gbpservice/nfp/configurator/agents/agent_base.py

```diff
@@ -143,45 +153,47 @@ def process_batch(self, ev):
                 # Process the first data blob from the request list.
                 # Get necessary parameters needed for driver method invocation.
                 method = request['method']
-                resource = request['resource']
-                kwargs = request['kwargs']
-                request_info = kwargs['kwargs']['request_info']
-                del kwargs['kwargs']['request_info']
-                context = request['context']
-                service_type = kwargs.get('kwargs').get('service_type')
+                is_generic_config = request['is_generic_config']
+                resource_data = request['resource_data']
+                agent_info = request['agent_info']
+                resource = agent_info['resource']
+                context = agent_info['context']
```

Comment:
Lets add a comment telling if this is neutron context or API context. Do this in all places where it is referred.